decidim-core/app/jobs/decidim/delete_inactive_participants_job.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module Decidim
+  # A job to delete inactive participants and notify them
+  class DeleteInactiveParticipantsJob < ApplicationJob
+    queue_as :delete_inactive_participants
+
+    def perform(organization)
+      clear_notifications_for_active_users(organization)
+      send_first_warning_notifications(organization.users.first_warning_inactive_users)
+      send_last_warning_notifications(organization.users.last_warning_inactive_users)
+      remove_inactive_users(organization.users.removable_users)
+    end
+
+    private
+
+    def clear_notifications_for_active_users(organization)
+      process_users(organization.users.not_deleted
+                                .with_inactivity_notification
+                                .active_after_notification) do |user|
+        user.extended_data.delete("inactivity_notification")
+        user.save!
+      end
+    end
+
+    def send_first_warning_notifications(users)
+      send_warning_notification(users, "first", Decidim.delete_inactive_users_first_warning_days_before)
+    end
+
+    def send_last_warning_notifications(users)
+      send_warning_notification(users, "second", Decidim.delete_inactive_users_last_warning_days_before)
+    end
+
+    def send_warning_notification(users, type, days)
+      process_users(users) do |user|
+        user.update!(
+          extended_data: user.extended_data.deep_merge!(
+            "inactivity_notification" => {
+              "notification_type" => type,
+              "sent_at" => Time.current
+            }
+          )
+        )
+        user.save!
+        send_notification(user, :inactivity_notification, days)
+      end
+    end
+
+    def remove_inactive_users(users)
+      process_users(users) do |user|
+        send_notification(user, :removal_notification, user.email, user.name, user.organization)
```

Comment:
I would use the user locale as well for this. There are installations using multiple languages. 

```suggestion
        send_notification(user, :removal_notification, user.email, user.name, user.locale, user.organization)
```